### PR TITLE
feat(askar): improved hardware key support

### PR DESCRIPTION
**Pull request by berendsliedrecht**
Signed-off-by: Berend Sliedrecht <sliedrecht@berend.io>

### CODE REVIEW

File: packages/askar/src/wallet/AskarBaseWallet.ts [hunk 1]
```diff
@@ -181,6 +191,22 @@ export abstract class AskarBaseWallet implements Wallet {
...
+        const publicKeyBytes = secureEnvironment.getPublicBytesForKeyId(kid)
+        const publicKeyBase58 = TypedArrayEncoder.toBase58(publicKeyBytes)
+
+        await this.storeHardwareKeyById({
+          keyType,
+          publicKeyBase58,
+          keyId: kid,
+        })
+
+        return new Key(publicKeyBytes, keyType, keyId)
```
Comment (berendsliedrecht) [inline packages/askar/src/wallet/AskarBaseWallet.ts:209]: Should be `kid` here

File: packages/askar/src/wallet/AskarBaseWallet.ts [hunk 2]
```diff
@@ -482,6 +518,18 @@ export abstract class AskarBaseWallet implements Wallet {
     }
   }
 
+  private async doesSecureEnvironmentKeyExist(keyId: string): Promise<boolean> {
+    try {
+      const entryObject = await this.withSession((session) =>
+        session.fetch({ category: 'SecureEnvironmentRecord', name: keyId })
```
Comment (TimoGlastra) [inline packages/askar/src/wallet/AskarBaseWallet.ts:524]:
```suggestion
        session.fetch({ category: 'SecureEnvironmentKeyRecord', name: keyId })
```

File: packages/askar/src/wallet/AskarBaseWallet.ts [hunk 3]
```diff
@@ -509,4 +557,28 @@ export abstract class AskarBaseWallet implements Wallet {
       throw new WalletError('Error saving KeyPair record', { cause: error })
     }
   }
+
+  private async storeHardwareKeyById(options: {
```
Comment (TimoGlastra) [inline packages/askar/src/wallet/AskarBaseWallet.ts:561]: incosistent naming hardwareKey vs secureEnvironmentKey

File: packages/core/src/crypto/Key.ts [hunk 4]
```diff
@@ -8,10 +8,12 @@ import { getKeyTypeByMultiCodecPrefix, getMultiCodecPrefixByKeyType } from './mu
 export class Key {
   public readonly publicKey: Buffer
   public readonly keyType: KeyType
+  public keyId: string
```
Comment (TimoGlastra) [inline packages/core/src/crypto/Key.ts:11]:
```suggestion
  /** the identifier of the key. If not provided in the constructor the base58 encoded public key will be used as the key identifier by default
  */
  public keyId: string
```

**Review by TimoGlastra — APPROVED** [review]
This looks good to me 👍 Will you remove the secure env integration from askar again?

**Pull request merged**